piptools/utils.py

```diff
@@ -696,35 +699,48 @@ def parse_config_file(
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
-    cli_opts = get_cli_options(click_context)
-
     # In a TOML file, we expect the config to be under `[tool.pip-tools]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})
 
     # Replace boolean flags like ``--no-annotate`` with their equivalents
-    try:
-        piptools_config = {
-            cli_opts["--" + k].name
-            if ("--" + k) in cli_opts.keys()
-            else k.lstrip("-")
-            .replace("-", "_")
-            .lower(): (
-                not v
-                if k.startswith("no-") and not k == "no-index" and isinstance(v, bool)
-                else v
-            )
-            for k, v in piptools_config.items()
-        }
-    except KeyError:
-        pass
+    config = _invert_negative_bool_options(
+        ctx=click_context,
+        config=piptools_config,
+    )
+
     # Any option with multiple values needs to be a list in the pyproject.toml
     for mv_option in MULTIPLE_VALUE_OPTIONS:
-        if not isinstance(piptools_config.get(mv_option), (list, type(None))):
+        if not isinstance(config.get(mv_option), (list, type(None))):
             original_option = mv_option.replace("_", "-")
             raise click.BadOptionUsage(
                 original_option, f"Config key '{original_option}' must be a list"
             )
-    return piptools_config
+    return config
+
+
+def _invert_negative_bool_options(
+    ctx: click.Context, config: dict[str, Any]
+) -> dict[str, Any]:
+    new_config = {}
+    cli_opts = get_cli_options(ctx)
+    for key, value in config.items():
+        long_option = "--" + key
+        option_name = cli_opts[long_option].name
+        assert option_name is not None
```

Comment:
```suggestion
        option_name = cli_opts.get(long_option)
        assert option_name is not None
        option_name = option_name.name
```

Reply:
Fixed in 4ee4de3ba11a41f86b70bedb1c41d2fdce1ab6e4